### PR TITLE
Merge main into previews/v2 part 1

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Merging main into previews/v2 would be massive, so I'm taking this a few commits at a time

### CODE REVIEW

**Review by brycewang-microsoft — APPROVED** [review]

File: e2e/test/docker/readme.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # Docker for WSL setup for local test
 
 ## Intro
-
```
Comment (drwill-ms) [inline e2e/test/docker/readme.md:4]: there are supposed to be newlines before/after headers and lists.

File: provisioning/device/samples/Getting Started/X509Sample/Parameters.cs [hunk 2]
```diff
@@ -41,6 +41,7 @@ internal class Parameters
         public string CertificatePassword { get; set; }
 
         [Option(
+            'g',
```
Comment (drwill-ms) [inline provisioning/device/samples/Getting Started/X509Sample/Parameters.cs:44]: Why give something that won't likely be changed a shortcut?

Our take has generally evolved to only having aliases for the required/commonly changed params.

**Review by drwill-ms — APPROVED** [review]

**Comment by drwill-ms** [comment]
My comments about the code changes have more to do with how they are in main, and not about the merge commit.

**Pull request merged**